AccessHandle.md

```diff
@@ -0,0 +1,247 @@
+    precondition to operate on recently written data.
+*   Its design ensures that implementations can access data from “the renderer
+    process” (i.e. without needing to rely on a component that coordinates
+    between execution contexts) to avoid costly inter-process communication
+    (IPCs). Examples of this approach include the allocation-based quota system
+    and only allowing one open file handle per file.
+*   We avoid extra memory allocations by taking a buffer when reading.
+
+In order to bring this use cases to OPFS, we propose adding a method to
+*FileSystemFileHandle* that returns an *AccessHandle*, a new interface.
+*AccessHandles* provide the right kind of access and locking semantics to achieve
+similar performance to Storage Foundation API.
+
+Further details on the origins of this proposal, and alternatives considered,
+can be found in the following documents:
+[Merging Storage Foundation API and the Origin Private File System](https://docs.google.com/document/d/121OZpRk7bKSF7qU3kQLqAEUVSNxqREnE98malHYwWec),
+[Recommendation for Augmented OPFS](https://docs.google.com/document/d/1g7ZCqZ5NdiU7oqyCpsc2iZ7rRAY1ZXO-9VoG4LfP7fM).
+
+## Goals and Use Cases
+
+Out goals and use cases are the same as
+[Storage Foundation API](https://github.com/WICG/storage-foundation-api-explainer),
+namely to give developers flexibility by providing generic, simple, and
+performant primitives upon which they can build higher-level components. The
+new surface is particularly well suited for Wasm-based libraries and
+applications that want to use custom storage algorithms to fine-tune execution
+speed and memory usage.
+
+A few examples of what could be done with *AccessHandles*:
+
+*   Allow tried and true technologies to be performantly used as part of web
+    applications e.g. using a port of your favorite storage library
+    within a website
+*   Distribute a Wasm module for WebSQL, allowing developers to us it across
+    browsers and opening the door to removing the unsupported API from Chrome
+*   Allow a music production website to operate on large amounts of media, by
+    relying on the new surface's performance and direct buffered access
+    to offload sound segments to disk instead of holding them in memory
+*   Provide a persistent [Emscripten](https://emscripten.org/) filesystem that
+    outperforms
+    [IDBFS](https://emscripten.org/docs/api_reference/Filesystem-API.html#filesystem-api-idbfs)
+    and has a simpler implementation
+
+## Non-goals
+
+This proposal is focused only on additions to the
+[Origin Private File System](https://wicg.github.io/file-system-access/#sandboxed-filesystem),
+and doesn't currently consider changes to the rest of File System Access API or
+how files in the host machine are accessed.
+
+## Proposed API
+
+### New data access surface
+
+```javascript
+//In all contexts
+const handle = await file.createAccessHandle();
+await handle.writable.getWriter().write(buffer);
```

Comment:
The "writable" here is a bit confusing, since it is not the writable that's created by "createWritable()", especially since these other writables are mentioned in Line 159. If you cannot rename them, maybe

Reply:
Sadly that name is part of the definition of a duplex stream, so there is benefit to keeping it. I think there's a part of yor comment missing, what is your suggestion?